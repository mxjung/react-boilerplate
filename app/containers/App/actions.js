/*
 * App Actions
 *
 * Actions change things in your application
 * Since this boilerplate uses a uni-directional data flow, specifically redux,
 * we have these actions which are the only way your application interacts with
 * your application state. This guarantees that your state is up to date and nobody
 * messes it up weirdly somewhere.
 *
 * To add a new Action:
 * 1) Import your constant
 * 2) Add a function like this:
 *    export function yourAction(var) {
 *        return { type: YOUR_ACTION_CONSTANT, var: var }
 *    }
 */

import {
  LOAD_INPUTS,
  LOAD_INPUTS_SUCCESS,
  LOAD_INPUTS_ERROR,
  POST_INPUT,
} from './constants';

/**
 * Load the user inputs, this action starts the request saga in Home
 *
 * @return {object} An action object with a type of LOAD_INPUTS
 */
export function loadInputs() {
  return {
    type: LOAD_INPUTS,
  };
}

/**
 * post new user input, this action starts the request saga in InputEntryPage
 *
 * @return {object} An action object with a type of POST_INPUT
 */
export function postInput() {
  return {
    type: POST_INPUT,
  };
}

/**
 * Dispatched when the user inputs are loaded by the request saga
 *
 * @param  {array} inputs The array containing user input strings
 *
 * @return {object}      An action object with a type of LOAD_INPUTS_SUCCESS passing the inputs
 */
export function inputsLoaded(inputs) {
  return {
    type: LOAD_INPUTS_SUCCESS,
    inputs,
  };
}

/**
 * Dispatched when loading the inputs fails
 *
 * @param  {object} error The error
 *
 * @return {object}       An action object with a type of LOAD_REPOS_ERROR passing the error
 */
export function inputLoadingError(error) {
  return {
    type: LOAD_INPUTS_ERROR,
    error,
  };
}
