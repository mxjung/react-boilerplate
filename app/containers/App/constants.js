/*
 * AppConstants
 * Each action has a corresponding type, which the reducer knows and picks up on.
 * To avoid weird typos between the reducer and the actions, we save them as
 * constants here. We prefix them with 'yourproject/YourComponent' so we avoid
 * reducers accidentally picking up actions they shouldn't.
 *
 * Follow this format:
 * export const YOUR_ACTION_CONSTANT = 'yourproject/YourContainer/YOUR_ACTION_CONSTANT';
 */

export const LOAD_REPOS = 'boilerplate/App/LOAD_REPOS';
export const LOAD_REPOS_SUCCESS = 'boilerplate/App/LOAD_REPOS_SUCCESS';
export const LOAD_REPOS_ERROR = 'boilerplate/App/LOAD_REPOS_ERROR';

// mxjung
export const LOAD_INPUTS = 'boilerplate/App/LOAD_INPUTS';
export const LOAD_INPUTS_SUCCESS = 'boilerplate/App/LOAD_INPUTS_SUCCESS';
export const LOAD_INPUTS_ERROR = 'boilerplate/App/LOAD_INPUTS_ERROR';
export const POST_INPUT = 'boilerplate/App/POST_INPUT';
