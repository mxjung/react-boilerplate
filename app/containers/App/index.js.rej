diff a/app/containers/App/index.js b/app/containers/App/index.js	(rejected hunks)
@@ -20 +20 @@ import Header from 'components/Header';
-import Footer from 'components/Footer';
+// import Footer from 'components/Footer';
