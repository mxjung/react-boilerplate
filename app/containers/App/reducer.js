/*
 * AppReducer
 *
 * The reducer takes care of our data. Using actions, we can
 * update our application state. To add a new action,
 * add it to the switch statement in the reducer function
 *
 */

import produce from 'immer';
import {
  LOAD_INPUTS,
  LOAD_INPUTS_SUCCESS,
  LOAD_INPUTS_ERROR,
} from './constants';

// The initial state of the App
export const initialState = {
  loading: false,
  error: false,
  userInputs: [],
};

/* eslint-disable default-case, no-param-reassign */
const appReducer = (state = initialState, action) =>
  produce(state, draft => {
    switch (action.type) {
      case LOAD_INPUTS:
        draft.loading = true;
        draft.error = false;
        draft.userInputs = [];
        break;

      case LOAD_INPUTS_SUCCESS:
        draft.userInputs = action.inputs;
        draft.loading = false;
        break;

      case LOAD_INPUTS_ERROR:
        draft.error = action.error;
        draft.loading = false;
        break;
    }
  });

export default appReducer;
