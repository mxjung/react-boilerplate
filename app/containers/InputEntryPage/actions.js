/*
 * InputEntryPage Actions
 *
 * Actions change things in your application
 * Since this boilerplate uses a uni-directional data flow, specifically redux,
 * we have these actions which are the only way your application interacts with
 * your application state. This guarantees that your state is up to date and nobody
 * messes it up weirdly somewhere.
 *
 * To add a new Action:
 * 1) Import your constant
 * 2) Add a function like this:
 *    export function yourAction(var) {
 *        return { type: YOUR_ACTION_CONSTANT, var: var }
 *    }
 */

import {
  CHANGE_INPUT,
  RESET_INPUT,
  VALID_INPUT,
  ERROR_MESSAGE,
} from './constants';

/**
 * Changes the input field value of the form
 *
 * @param  {string} input The new text of the input field
 *
 * @return {object} An action object with a type of CHANGE_INPUT and input value
 */
export function changeInput(input) {
  return {
    type: CHANGE_INPUT,
    input,
  };
}

/**
 * Dispatched to erase user input from form
 *
 * @return {object}      An action object with a type of RESET_INPUT
 */
export function resetInput() {
  return {
    type: RESET_INPUT,
  };
}

/**
 * Dispatched to change state of whether user input is valid or not
 *
 * @param  {array} valid Boolean for whether input is valid or not
 *
 * @return {object}      An action object with a type of VALID_INPUT and a valid boolean
 */
export function toggleValidInput(valid) {
  return {
    type: VALID_INPUT,
    valid,
  };
}

/**
 * Dispatched to update error msg to notify string that surpasses threshold length
 *
 * @param  {string} errorMsg  Error string message
 *
 * @return {object}           An action object with a type of ERROR_MESSAGE and a errorMsg
 */
export function changeErrorMsg(errorMsg) {
  return {
    type: ERROR_MESSAGE,
    errorMsg,
  };
}
