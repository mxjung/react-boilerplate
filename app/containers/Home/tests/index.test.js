/**
 * Test the HomePage
 */

import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';

import { Home, mapDispatchToProps } from '../index';
// import { changeUsername } from '../actions';
import { loadInputs } from '../../App/actions';
import configureStore from '../../../configureStore';

describe('<Home />', () => {
  let store;

  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <IntlProvider locale="en">
          <Home
            loading={false}
            error={false}
            inputs={[]}
            dispatchInputs={jest.fn()}
          />
        </IntlProvider>
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });

  it('should call dispatchInputs to fetch the user inputs on first mount', () => {
    const submitSpy = jest.fn();
    render(
      <Provider store={store}>
        <IntlProvider locale="en">
          <Home inputs={[]} dispatchInputs={submitSpy} />
        </IntlProvider>
      </Provider>,
    );
    expect(submitSpy).toHaveBeenCalled();
  });

  it('should not call dispatchInputs if inputs is not an empty array', () => {
    const submitSpy = jest.fn();
    render(
      <Provider store={store}>
        <IntlProvider locale="en">
          <Home inputs={['test']} dispatchInputs={submitSpy} />
        </IntlProvider>
      </Provider>,
    );
    expect(submitSpy).not.toHaveBeenCalled();
  });

  describe('mapDispatchToProps', () => {
    describe('dispatchInputs', () => {
      it('should be injected', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        expect(result.dispatchInputs).toBeDefined();
      });

      it('should dispatch loadInputs when called', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        result.dispatchInputs();
        expect(dispatch).toHaveBeenCalledWith(loadInputs());
      });
    });
  });
});
