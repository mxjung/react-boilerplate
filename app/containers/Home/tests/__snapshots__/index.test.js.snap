// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should render and match the snapshot 1`] = `
<article>
  <div>
    <section
      class="Section-eaizcf-0 CenteredSection-sc-4fyhoo-0 bKZikw"
    >
      <p>
        <span>
          There are no inputs. Add values by clicking on "Add String" button
        </span>
      </p>
    </section>
  </div>
</article>
`;
